src/core/action-creator.ts

```diff
@@ -51,7 +51,33 @@ export function createActions(flux: FluxCapacitor) {
                 func: ({ query }) => !('query' in search) || !!query || query === null,
                 msg: 'search term is empty'
               }, {
-                func: ({ query }, state) => query !== Selectors.query(state) || query === null,
+                  // currently assume that search will only ever have up to one refinement
+                  func: ({ query, value, navigationId, range, low, high, index }, state) => {
```

Comment:
could be:

```js
import SearchAdapter from './adapters/search';

function refinementsChanged(clear, navigationId, index, value, low, high, state) {
  const currentRefinements = Selectors.selectedRefinements(state);
  const hasIndex = index >= 0;
  if (clear) {
    const clearedNavigation = Selectors.navigation(state, clear);
    if (clearedNavigation) {
      if (clear === navigationId) {
        if (clearedNavigation.refinements.length !== 1) {
          return true;
        } else if (hasIndex) {
          return isRefinementDeselected(state, navigationId, index);
        } else {
          return SearchAdapter.refinementsMatch({ low, high, value }, clearedNavigation.refinements[0], clearedNavigation.range || 'Value');
        }
      } else {
        return clearedNavigation.refinements.length !== 0;
      }
    } else if (currentRefinements.length === 1) {
      if (hasIndex) {
        return isRefinementDeselected(state, navigationId, index);
      } else {
        return SearchAdapter.refinementsMatch({ low, high, value }, currentRefinements[0]);
      }
    } else {
      return currentRefinements.length !== 0;
    }
  } else if (hasIndex) {
    return isRefinementDeselected(state, navigationId, index);
  } else {
    return !currentRefinements.find((refinement) => {
      return refinement.navigationName === navigationId && SearchAdapter.refinementsMatch({ low, high, value }, refinement);
    });
  }
}

func: ({ query, value, navigationId, low, high, index, clear }, state) => {
  const queryHasChanged = query !== Selectors.query(state);
  const refinementsHaveChanged = (navigationId || clear) && refinementsChanged(clear, navigationId, index, value, low, high, state);

  return queryHasChanged || refinementsHaveChanged;
}
```

Reply:
as you'll notice above, we still need to handle the case where we pass `clear` to indicate that we intend to clear all the refinements first, `clear` can be a boolean or the field of a `navigation`

Reply:
I plan to simplify the above action creator to turn it into a bulk action creator, I'll sit with you in the afternoon to go over updating it and the validator